internal/types/subtypes/attribute_transform.go

```diff
@@ -112,3 +112,57 @@ func protoToStruct(p proto.Message) (*structpb.Struct, error) {
 	}
 	return st, nil
 }
+
+// Filterable converts a proto.Message so any subtype attributes fields are
+// structed like the API so filter strings will be correctly applied. If the
+// given proto.Message does not have any subtype attributes, the original
+// proto.Message is returned.
+func Filterable(item proto.Message) (proto.Message, error) {
+	clone := proto.Clone(item)
+	r := clone.ProtoReflect()
+
+	attrsField := r.Descriptor().Oneofs().ByName("attrs")
```

Comment:
Add comments to underscore all the assumptions about field names and structure, please.

Reply:
See [24fe7b8](https://github.com/hashicorp/boundary/pull/1967/commits/24fe7b8a7acdd5176c291e72452144c3005a260e)